src/pwiz/data/msdata/IO.cpp

```diff
@@ -463,15 +463,16 @@ PWIZ_API_DECL void write(minimxml::XMLWriter& writer, const FileDescription& fd)
     write(writer, fd.fileContent);
 
     XMLWriter::Attributes attributes;
-    attributes.add("count", fd.sourceFilePtrs.size());
-    writer.startElement("sourceFileList", attributes);
-
-    for (vector<SourceFilePtr>::const_iterator it=fd.sourceFilePtrs.begin(); 
-         it!=fd.sourceFilePtrs.end(); ++it)
-         write(writer, **it);
-
-    writer.endElement();
-
+    if (fd.sourceFilePtrs.size() > 0) {
```

Comment:
This ensures that no empty `sourceFileList` element is written. An empty `sourceFileList` is not valid according to the mzML standard.